### PR TITLE
ipc4: dma-trace: fix ipc4 log buffer status notification handling

### PR DESCRIPTION
**Pull request by rfredzim**
Fix the IPC4 log buffer notification by
- using appropriate IPC header 
- decoupling dma-trace msg structure  from ipc3 specific definition
- minimize ratio of triggering dma transport

### CODE REVIEW

**Review by mmaka1 — APPROVED** [review]

**Comment by gkbldcig** [comment]
Can one of the admins verify this patch?

**Comment by rfredzim** [comment]
Please hold on with merge, since enabling the log buffer status seems to affect some of the local pipeline related tests with zephyr configuration.

File: src/ipc/ipc4/helper.c [hunk 1]
```diff
@@ -49,6 +49,13 @@ void ipc_build_comp_event(struct sof_ipc_comp_event *event, uint32_t type,
 {
 }
 
+bool ipc_trigger_trace_xfer(uint32_t avail)
+{
+	if (avail < DMA_TRACE_LOCAL_SIZE / 2)
+		return false;
+	return true;
```
Comment (lyakh) [inline src/ipc/ipc4/helper.c:56]: `return avail >= DMA_TRACE_LOCAL_SIZE / 2;

File: src/trace/dma-trace.c [hunk 2]
```diff
@@ -154,7 +154,7 @@ int dma_trace_init_early(struct sof *sof)
 
 	ipc_build_trace_posn(&sof->dmat->posn);
 	sof->dmat->msg = ipc_msg_init(sof->dmat->posn.rhdr.hdr.cmd,
-				      sizeof(sof->dmat->posn));
+			sof->dmat->posn.rhdr.hdr.size);
```
Comment (lgirdwood) [inline src/trace/dma-trace.c:160]: indentation needs to align with the ```ipc_msg_init(```

**Review by lgirdwood — COMMENTED** [review]
@rfredzim thanks, fwiw we will need to remember this when transferring the DMA trace to Zephyr.

**Comment by lgirdwood** [comment]
> Please hold on with merge, since enabling the log buffer status seems to affect some of the local pipeline related tests with zephyr configuration.

ok, please ping when it's ready.

**Comment by rfredzim** [comment]
> > Please hold on with merge, since enabling the log buffer status seems to affect some of the local pipeline related tests with zephyr configuration.
> 
> ok, please ping when it's ready.

It's ready, after recent rebase the local issue was gone.

**Comment by lgirdwood** [comment]
@tmleman @marcinszkudlinski @RanderWang any comments ?

**Review by lgirdwood — APPROVED** [review]

**Review by tmleman — APPROVED** [review]

**Pull request merged**